src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -0,0 +1,222 @@
+            }
+            if (isShutdown) {
+                return null;
+            }
+        }
+        log.info("Node bootstrapped. Layout received by management server.");
+        return currentLayout;
+    }
+
+    /**
+     * Returns a connected instance of the CorfuRuntime.
+     *
+     * @return A connected instance of runtime or null if isShutdown is true.
+     */
+    private synchronized CorfuRuntime getCorfuRuntime() {
+
+        if (corfuRuntime == null) {
+
+            // TODO: Check for persisted layout if saved by the management server.
+            // If yes, connect runtime to that layout as it is the latest snapshot
+            // of the layout servers we have.
+            // If no persisted layout present, wait for bootstrap. (Rely on local layout server)
+
+            Layout bootstrapLayout = waitForBootstrap();
+            if (bootstrapLayout == null) {
+                return null;
+            }
+
+            corfuRuntime = new CorfuRuntime();
+            bootstrapLayout.getLayoutServers().forEach(ls -> corfuRuntime.addLayoutServer(ls));
+            corfuRuntime.connect();
+        }
+        return corfuRuntime;
+    }
+
+
+    /**
+     * Gets the address of this endpoint.
+     *
+     * @return localEndpoint address
+     */
+    private String getLocalEndpoint() {
+        return this.opts.get("--address") + ":" + this.opts.get("<port>");
+    }
+
+
+    /**
+     * This contains the complete failure detection and handling mechanism.
+     * <p>
+     * It first checks whether the current node is bootstrapped.
+     * If not, it continues checking in intervals of 1 second.
+     * If yes, it sets up the corfuRuntime and continues execution
+     * of the policy.
+     * <p>
+     * It executes the policy which detects and reports failures.
+     * Once detected, it triggers the trigger handler which takes care of
+     * dispatching the appropriate handler.
+     * <p>
+     * Currently executing the periodicPollPolicy.
+     * It executes the the polling at an interval of every 1 second.
+     * After every poll it checks for any failures detected.
+     */
+    private void taskScheduler() {
+
+        Layout currentLayout;
+        LayoutView layoutView;
+        CorfuRuntime corfuRuntime;
+
+        corfuRuntime = getCorfuRuntime();
+        if (corfuRuntime == null) return;
+
+        corfuRuntime.invalidateLayout();
+
+        // Fetch the latest layout view through the runtime.
+        layoutView = corfuRuntime.getLayoutView();
+        currentLayout = layoutView.getLayout();
+        //TODO: Store this layout on disk for persistence.
+
+        // Execute the failure detection policy once.
+        failureDetectorPolicy.executePolicy(currentLayout, corfuRuntime);
+
+        // Get the server status from the policy and check for failures.
+        HashMap<String, Boolean> map = failureDetectorPolicy.getServerStatus();
+        if (!map.isEmpty()) {
```

Comment:
Shouldn't this code belong in the policy itself? how does the management server know how to interpret these values? I feel that this logic shouldnt leak to the management server.